File: Card.java
public class Card {
    private String name;
    private String type;
    private boolean isIn;

    public Card(String name, String type) {
        this.name = name;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public void setIn(boolean isIn) {
        this.isIn = isIn;
    }
}

File: card.go
package main

type Card struct {
	name string
	cardType string
}

func NewCard(name, cardType string) *Card {
	return &Card{name: name, cardType: cardType}
}

func (c *Card) GetName() string {
	return c.name
}

func (c *Card) GetType() string {
	return c.cardType
}